patches/tModLoader/Terraria/Program.TML.cs

```diff
@@ -169,8 +169,28 @@ private static void PortFilesFromXtoY(string superSavePath, string source, strin
 				lastLaunchedTml = (string)lastLaunchedTmlObject;
 		}
 		catch (Exception e) {
-			e.HelpLink = "https://github.com/tModLoader/tModLoader/wiki/Basic-tModLoader-Usage-FAQ#configjson-corrupted";
-			ErrorReporting.FatalExit($"Attempt to Port from \"{oldFolderPath}\" to \"{newFolderPath}\" aborted, the \"{sourceFolderConfig}\" file is corrupted.", e);
+			if (File.GetLastWriteTime(sourceFolderConfig) > migrationDay) {
+				// If the file was edited recently, we assume that an updated tModLoader edited it. This should ignore modifications made long ago by pre-migration logic tModLoader releases.
+				lastLaunchedTml = BuildInfo.tMLVersion.ToString();
+			}
+			else {
+				e.HelpLink = "https://github.com/tModLoader/tModLoader/wiki/Basic-tModLoader-Usage-FAQ#configjson-corrupted";
+				ErrorReporting.FatalExit($"Attempt to Port from \"{oldFolderPath}\" to \"{newFolderPath}\" aborted, the \"{sourceFolderConfig}\" file is corrupted.", e);
+			}
+		}
+
+		if (string.IsNullOrEmpty(lastLaunchedTml)) {
```

Comment:
Why have this `else` block and not show the choice box in case the config *is* corrupted? Why not auto-delete it at that point too?

Reply:
Added. No need to auto-delete, it'll be replaced during game loading anyway.